samples/SocialSample/Startup.cs

```diff
@@ -280,16 +279,18 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
                     {
                         // By default the client will be redirect back to the URL that issued the challenge (/login?authtype=foo),
                         // send them to the home page instead (/).
-                        await context.Authentication.ChallengeAsync(authType, new AuthenticationProperties() { RedirectUri = "/" });
+                        await context.ChallengeAsync(authType, new AuthenticationProperties() { RedirectUri = "/" });
                         return;
                     }
 
                     context.Response.ContentType = "text/html";
                     await context.Response.WriteAsync("<html><body>");
                     await context.Response.WriteAsync("Choose an authentication scheme: <br>");
-                    foreach (var type in context.Authentication.GetAuthenticationSchemes())
+                    var schemeProvider = context.RequestServices.GetRequiredService<IAuthenticationSchemeProvider>();
+                    foreach (var provider in await schemeProvider.GetAllSchemesAsync())
                     {
-                        await context.Response.WriteAsync("<a href=\"?authscheme=" + type.AuthenticationScheme + "\">" + (type.DisplayName ?? "(suppressed)") + "</a><br>");
+                        // REVIEW: we lost access to display name (which is buried in the handler options)
```

Comment:
Yeah its fallen into a gap, since they hang off the scheme options, but those are configured too late, the schemes need to be registered as part of ConfigureServices.  We can just have the handler set the schemes display name after its options are initialized i guess